sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ImplementIDisposableCorrectly.cs

```diff
@@ -283,7 +283,7 @@ private bool HasVirtualDisposeBool(INamedTypeSymbol typeSymbol)
                     .SelectMany(m => m.DeclaringSyntaxReferences)

```

Comment:
I think this line is causing the underlying issue: in the IDE, it looks like _DeclaringSyntaxReferences_ can return a syntax tree from a different project i.e. a different compilation (I'm guessing this can't happen when building in MSBuild since only the files in the current project are available).

However, I don't understand why this code is necessary - we've fetched a list of method symbols, checked _IsDisposeBool_... ...then we're trying to resolve the method symbols again, using _DeclaringSyntaxReferences_, just to check whether the method symbols is abstract or not. Why can't we just check the method symbols we already have?